### ISSUE TITLE
 Alcinoe Library in Delphi Design View

### ISSUE THREAD

**Issue by rtomazini42** [MagicFoundation/Alcinoe#294]
Subject: Issue with Alcinoe Library in Delphi Design View

Hi,

I'm facing an issue with the Alcinoe library in Delphi's design view. Although I am able to import Alcinoe successfully in my Pascal code, the components from Alcinoe are not appearing in the design view. I have tried running "Compile All" to see if it would create any components, but whenever I try to run it, either the "Init" command is not recognized by the Windows prompt (even though it's added to my environment variables), or it just displays the message "Press any key to continue".

I followed the installation steps mentioned in the documentation and ran the RunTests, which also pointed out the error "'..\InitEnvironment.bat' is not recognized as an internal or external command, operable program, or batch file."

I would appreciate any assistance or suggestions on how to resolve this issue and successfully add the Alcinoe components to my Delphi IDE. Here are some additional details:

-     Operating System: Windows 10
-     Delphi Version: 11.3 Alexandria
-     Installation location: Drive D (C drive is used for the operating system)

Thank you in advance for your help.

**Comment by Zeus64** [comment]
To see the component in the design palette you must in delphi select component > install package > add and then select Alcinoe\Libraries\bpl\Alcinoe\Win32\Alexandria\AlcinoeAlexandria.bpl (this bpl is build via the "Compile All" command

**Comment by rtomazini42** [comment]
Thank you for the response. However, it doesn't seem to compile as shown in the image below. It simply finishes and prompts to 'press any key'. In the tests, it doesn't find "'..\InitEnvironment.bat'", even though it's already added to the environment variables. Is there any other way to run "Compile All"? It has been attempted through the command prompt, PowerShell, and Bash, but none of them worked.
![image](https://github.com/MagicFoundation/Alcinoe/assets/18154642/9dace6ce-0965-4560-b1bd-20be14b36e82)

**Comment by Zeus64** [comment]
InitEnvironment.bat is in the same directory as CompileAll.bat. If CompileAll.bat can not find it then their is a big problem on your system :( in compileAll.bat it's look for .\InitEnvironment.bat not ..\InitEnvironment.bat are you sure you use the last version of alcinoe ?

**Comment by rtomazini42** [comment]
Yes, I am using the latest version, it is in the same folder, and I have tried other execution methods as well. I think I just got unlucky with the operating system haha. Apparently, it finds the .\InitEnvironment.bat, but it stops the execution from that point on. I modified the runtests to do that, and it seems that the same thing happens. It searches for InitEnvironment.bat and finds it, but it stops execution at that exact moment.

**Comment by rtomazini42** [comment]
Sorry for the delay in the response.
Well, the difficult part seems to be done. I'm not very good with the command prompt, so I did it line by line. The error was occurring in the line:
set ALDelphiDir=%PROGRAMFILES(X86)%\Embarcadero\Studio%ALDelphiVersion%
CALL "%ALDelphiDir%\bin\rsvars.bat"
IF ERRORLEVEL 1 goto ERROR

It seems that ALDelphiDir was always pointing to the C drive, as I mentioned earlier, which belongs only to the operating system. So I modified it with the direct path:
```
set ALDelphiDir=D:\Program Files (x86)\Embarcadero\Studio\22.0\bin
CALL "rsvars.bat"
IF ERRORLEVEL 1 goto ERROR
```


This way, the script worked!


But now another error, but this time much more understandable, and I'll try to fix it myself. "'"D:\Components\Alcinoe\Tools`\UnitNormalizer\UnitNormalizer.exe"' is not recognized as an internal command"`

As soon as I fix it, I'll close the issue with the solution for this additional error. For now, in this gist, you can find the record of the modified Initialize file and the new error: https://gist.github.com/rtomazini42/63b691504c298422a4910fb32f089038

Thank you for all the help so far.

**Comment by Zeus64** [comment]
Oh I understand you didn't install delphi under the %PROGRAMFILES(X86)% 
I will try to update the .bat command to retrieve the path of the delphi from the regitry instead

regarding "D:\Components\Alcinoe\Tools\UnitNormalizer\UnitNormalizer.exe" you must build first UnitNormalizer.exe, but normally the compileAll.bat do it for you

**Comment by rtomazini42** [comment]
Well, apparently, since I don't have the command line compiler in the system, I compiled everything that was missing and kept the .exe in the indicated folder, finally the BPL too, and in the end, I installed the component using the BPL package produced in RAD itself. Everything turned out fine in the end, a bit more laborious but it happens.

Thank you very much for accompanying me, Zeus64, and for updating and providing support for the package. Have a great week!
